src/main/resources/testchipip/csrc/testchip_tsi.cc

```diff
@@ -3,10 +3,22 @@
 testchip_tsi_t::testchip_tsi_t(int argc, char** argv, bool can_have_loadmem) : tsi_t(argc, argv)
 {
   has_loadmem = false;
+  init_writes = std::vector<std::pair<uint64_t, uint32_t>>();
+
   std::vector<std::string> args(argv + 1, argv + argc);
   for (auto& arg : args) {
     if (arg.find("+loadmem=") == 0)
       has_loadmem = can_have_loadmem;
+    if (arg.find("+init_write=") == 0) {
+      auto d = arg.find(":");
+      if (d == std::string::npos) {
+	throw std::invalid_argument("Improperly formatted +init_write argument");
+      }
+      uint64_t addr = strtoull(arg.substr(12, d - 12).c_str(), 0, 16);
+      uint32_t val = strtoull(arg.substr(d + 1).c_str(), 0, 16);
```

Comment:
This enforces that you can only put hex numbers, that's fine but can that be documented somewhere or maybe have an extra check to ensure there is a `0x` prefix.